### PR TITLE
Updating compilers

### PR DESCRIPTION
**Pull request by lucamar**
Cray, GNU, Intel and PGI compilers have now separate pages

### CODE REVIEW

File: scientific_computing/compiling_your_code/pgi/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+The `-fast` (`-fastsse`) flag turns on optimizations relevant for the target platform and optimization level to a minimum of `-O2`.
+
+More aggressive optimization can be obtained by adding `-O3`, ie:
+
+* `-O3 -fast`
+
+At the `-O3` level, all level `2` optimizations are performed, and in addition, more aggressive code hoisting and scalar replacement optimizations are performed. These optimizations may speed up your code but might also slow it down, so it is always recommended to benchmark the performance of your code with a variety of options enabled/disabled. It may be worth experimenting with the `-Munroll`, `-Minline`, `-Mmovnt` and `-Mconcur` options in particular. Use `-help` to list the compiler options available or to see details on how to use a given option, e.g. `pgf95 -Munroll -help`.
+
+On the Cray systems, optimizations targeting the specific compute node architecture chosen by loading the `daint-gpu` or `daint-mc` module as discussed in [Compiling Your Code](http://eth-cscs.github.io/compiling_and_optimizing/compiling_your_code/#) section (Specific note for the hybrid Piz Daint XC50/XC40) are turned on by default.
```
Comment (victorusu) [inline scientific_computing/compiling_your_code/pgi/README.md]: what do you think about using relative links here?

File: scientific_computing/compiling_your_code/intel/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+The default optimization level with the Intel compiler (`-O2`) is quite high, so we suggest starting with the default optimization.
+
+It is worth testing the performance with `-O3`, which turns on more aggressive loop optimizations (loop unrolling, scalar replacement, branch elimination etc). Inter-procedural optimization can be switched on with `-ipo`, but as ever, you should benchmark your application to see if it benefits from IPO or not.
+
+We suggest trying `-O3 -unroll-aggressive -opt-prefetch` or fine tuning these parameters individually - the number of times to unroll loops (`-unroll[n]`), and the level of software prefetching (`-opt-prefect=n`).
+
+There is a useful flag `-opt-report[n]` (`n=0-3`, where `3` refers to most verbose) which will generate an optimization report, which is written to standard error.
+
+On the Cray systems, optimizations targeting the specific compute node architecture chosen by loading the `daint-gpu` or `daint-mc` module as discussed in [Compiling Your Code](http://eth-cscs.github.io/compiling_and_optimizing/compiling_your_code/#) section (Specific note for the hybrid Piz Daint XC50/XC40) are turned on by default.
```
Comment (victorusu) [inline scientific_computing/compiling_your_code/intel/README.md]: what do you think about using relative links here?

File: scientific_computing/compiling_your_code/gnu/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+Note that loop unrolling is not turned on by default at `-O3`, hence the need to include a specific flag for unrolling. Useful options to improve performance are:
+
+* `-funroll-loops` to unroll loops where the number of iterations is known at compiler time or on entry to the loop
+* `-funroll-all-loops` to unroll all loops (may make code run slower!)
+* `-ffast-math` (see [Floating point accuracy](#floating-point-accuracy) below)
+* `-finline-functions` to integrate simple functions into their callers (default at `-O3`)
+* `-finline-limit=n` to limit the size of functions inlined
+
+On the Cray systems, optimizations targeting the specific compute node architecture chosen by loading the `daint-gpu` or `daint-mc` module as discussed in [Compiling Your Code](http://eth-cscs.github.io/compiling_and_optimizing/compiling_your_code/#) section (Specific note for the hybrid Piz Daint XC50/XC40) are turned on by default. On non-Cray systems we recommend adding the `-march=native` flag.
```
Comment (victorusu) [inline scientific_computing/compiling_your_code/gnu/README.md]: what do you think about using relative links here?

**Comment by lucamar** [comment]
Thanks @victorusu, I have corrected the links and they are now relative to the GitHub pages.

**Pull request merged**